### ISSUE TITLE
on_done is not called with on_start args

### ISSUE THREAD

**Issue by kalkschneider** [cgarciae/pypeln#29]
Hello Cristian,

In your last release you changed the way the callback functions work. The return values of `on_start` are not passed to `on_done` as input arguments anymore. I hope you didn't do it on purpose, that makes it hard to close open connections if a worker has finished.

Your old code:

```python
args = params.on_start(worker_info)
params.on_done(stage_status, *args)
```

Your new code:

```python
f_kwargs = self.on_start(**on_start_kwargs)
on_done_kwargs = {}
done_resp = self.on_done(**on_done_kwargs)
```

**Comment by cgarciae** [comment]
Hey @kalkschneider!

Let me check. :open_mouth:  I think I have tests for this.

**Comment by cgarciae** [comment]
Fixed in version `0.2.5`.

**Comment by cgarciae** [comment]
@kalkschneider Thanks for reporting the issue! 
It was a serious bug resulting from the recent refactor.

BTW: Version `0.2` brought small modifications to how these aditional arguments are handled, instead of `*args` we are now using `**kwargs` which means that `on_start` now returns a dictionary. Take a look at the new documentation for the behavior: https://cgarciae.github.io/pypeln/advanced/#resource-management
